editor/src/components/canvas/controls/new-canvas-controls.tsx

```diff
@@ -269,15 +309,173 @@ const NewCanvasControlsInner = (props: NewCanvasControlsInnerProps) => {
     'NewCanvasControlsInner',
   )
 
-  const { localSelectedViews, localHighlightedViews, setLocalSelectedViews } = props
+  const {
+    localSelectedViews,
+    localHighlightedViews,
+    setLocalSelectedViews,
+    setLocalHighlightedViews,
+    setSelectionAreaRectangle,
+  } = props
   const cmdKeyPressed = keysPressed['cmd'] ?? false
 
   const contextMenuEnabled = !isLiveMode(editorMode)
   const { maybeHighlightOnHover, maybeClearHighlightsOnHoverEnd } = useMaybeHighlightElement()
 
-  const { onMouseMove, onMouseDown, onMouseUp } = useSelectAndHover(
-    cmdKeyPressed,
-    setLocalSelectedViews,
+  const selectModeHooks = useSelectAndHover(cmdKeyPressed, setLocalSelectedViews)
+
+  const ref = React.useRef<HTMLDivElement | null>(null)
+
+  const [mousePoint, setMousePoint] = React.useState<CanvasPoint | null>(null)
+  const [selectionAreaStart, setSelectionAreaStart] = React.useState<CanvasPoint | null>(null)
+
+  const storeRef = useRefEditorState((store) => {
+    return { jsxMetadata: store.editor.jsxMetadata }
+  })
+
+  const getCanvasPoint = React.useCallback(
+    (x: number, y: number): CanvasPoint => {
+      return windowToCanvasCoordinates(scale, canvasOffset, windowPoint({ x, y }))
+        .canvasPositionRounded
+    },
+    [scale, canvasOffset],
+  )
+
+  const mousePointOnCanvas = React.useMemo(() => {
+    if (mousePoint == null) {
+      return null
+    }
+    return getCanvasPoint(mousePoint.x, mousePoint.y)
+  }, [mousePoint, getCanvasPoint])
+
+  const selectionArea = React.useMemo((): CanvasRectangle | null => {
```

Comment:
I think `selectionArea` would probably work better as a `useState` (set via the mouse event handlers) than a `useMemo`

Reply:
makes total sense, I got rid of it and made it local to `onMouseMove` here https://github.com/concrete-utopia/utopia/pull/3750/commits/b342869dcca26246b84df8bb7caca117f9082fcc